library/src/main/java/com/datatorrent/lib/partitioner/ClonePartitioner.java

```diff
@@ -0,0 +1,143 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.partitioner;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.validation.constraints.Min;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import com.datatorrent.api.DefaultPartition;
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.Operator.InputPort;
+import com.datatorrent.api.Partitioner;
+
+/**
+ * This is a simple partitioner, which replicates data across all partitions of an operator.
+ *
+ * @param <T> The type of the operator
+ * @since 2.0.0
+ */
+public class ClonePartitioner<T extends Operator> implements Partitioner<T>, Serializable
+{
+  private static final Logger logger = LoggerFactory.getLogger(ClonePartitioner.class);
+  
+  private static final long serialVersionUID = 201462371710L;
+  /**
+   * The number of partitions for the default partitioner to create.
+   */
+  @Min(1)
+  private int partitionCount = 1;
+
+  /**
+   * This creates a partitioner which creates only one partition.
+   */
+  public ClonePartitioner()
+  {
+  }
+
+  /**
+   * This constructor is used to create the partitioner from a property.
+   * @param value A string which is an integer of the number of partitions to create
+   */
+  public ClonePartitioner(String value)
+  {
+    this(Integer.parseInt(value));
+  }
+
+  /**
+   * This creates a partitioner which creates partitonCount partitions.
+   * @param partitionCount The number of partitions to create.
+   */
+  public ClonePartitioner(int partitionCount)
+  {
+    this.partitionCount = partitionCount;
+  }
+
+  /**
+   * This method sets the number of partitions for the ClonePartitioner to create.
+   * @param partitionCount The number of partitions to create.
+   */
+  public void setPartitionCount(int partitionCount)
+  {
+    this.partitionCount = partitionCount;
+  }
+
+  /**
+   * This method gets the number of partitions for the ClonePartitioner to create.
+   * @return The number of partitions to create.
+   */
+  public int getPartitionCount()
+  {
+    return partitionCount;
+  }
+
+  @Override
+  public Collection<Partition<T>> definePartitions(Collection<Partition<T>> partitions, PartitioningContext context)
+  {
+    final int newPartitionCount = DefaultPartition.getRequiredPartitionCount(context, this.partitionCount);
+    logger.debug("define partitions, partitionCount current {} requested {}", partitions.size(), newPartitionCount);
+
+    //Get a partition
+    DefaultPartition<T> partition = (DefaultPartition<T>)partitions.iterator().next();
+    ArrayList<Partition<T>> newPartitions;
```

Comment:
why not `ArrayList<Partition<T>> newPartitions = Lists.newArrayList();` and remove line 112?